### ISSUE TITLE
brace-style allman ignores indent

### ISSUE THREAD

**Issue by ew72ew** [eslint/eslint#7662]
**Tell us about your environment**

* **ESLint Version:** 3.11.0
* **Node Version:** 6.2.1
* **npm Version:** 4.0.2

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**

```js
module.exports = {
    'env': {
        'es6': true,
        'browser': true,
    },

    'rules': {
        'indent': ['warn', 4],
        'brace-style': ['warn', 'allman'],
    }
};
```

**What did you do? Please include the actual source code causing the issue.**

```js
function test1()
{
    function test2()
{ // wrong indentation here
        function test3()
{ // and here
        }
    }
}
```

**What did you expect to happen?**

```
4:1  warning  Expected indentation of 4 spaces but found 0  indent
6:1  warning  Expected indentation of 8 spaces but found 0  indent
```

**What actually happened? Please include the actual, raw output from ESLint.**

No warnings displayed

**Comment by not-an-aardvark** [comment]
Thanks, I can reproduce this issue. It appears to be a bug in the `indent` rule.

I think this will be fixed after https://github.com/eslint/eslint/pull/7618 is merged.

**Comment by helgeh** [comment]
In my opinion this is not only an issue with the `indent` rule. It is more clearly a problem with `brace-style` if you go with the `stroustrup` setting. Then, when a brace is on the next line and the rule fixes it by removing the linefeed, all indentation is left as is in between keyword/parenthesis and the brace.

Result:
```javascript
    function fancyAlert(arg)     {
      if(arg)       {
        $.facebox({div:'#foo'});
      }
    }
```
